src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -1361,53 +1337,24 @@ public bool Translate(object sender, X509Certificate cert, X509Chain chain, SslP
         {
             add
             {
-                _serverCallbackRWLock.EnterWriteLock();
-                try
+                if (PlatformDetails.RunningOnMacOsx)
+                    throw new PlatformNotSupportedException("On Mac OSX, is it not possible to register to RemoteCertificateValidationCallback because of https://github.com/dotnet/corefx/issues/9728");
+
+                lock (_locker)
                 {
                     _serverCertificateCustomValidationCallback = _serverCertificateCustomValidationCallback.Concat(new[] { value }).ToArray();
-                    ForceUpdateOfAllHttpClients();
-                }
-                finally
-                {
-                    _serverCallbackRWLock.ExitWriteLock();
                 }
             }
+
             remove
             {
-                _serverCallbackRWLock.EnterWriteLock();
-                try
-                {
-                    _serverCertificateCustomValidationCallback = _serverCertificateCustomValidationCallback.Except(new[] { value }).ToArray();
-                    ForceUpdateOfAllHttpClients();
-                }
-                finally
-                {
-                    _serverCallbackRWLock.ExitWriteLock();
-                }
-            }
-        }
+                if (PlatformDetails.RunningOnMacOsx)
+                    throw new PlatformNotSupportedException("On Mac OSX, is it not possible to register to RemoteCertificateValidationCallback because of https://github.com/dotnet/corefx/issues/9728");
```

Comment:
I don't like this approach. It means that when OSX does support it, we'll have to do something special for this.

Reply:
Can you check now?